typescript/search-attributes.md

```diff
@@ -0,0 +1,86 @@
+## Search attributes V2
+
+### Why do we need this?
+
+In the current version of the SDK, search attributes are always arrays. In order for the server to support visibility
+on top of standard persistence, we need to redo the search attributes API to enforce single values for all search
+attributes except for keywords. The server will eventually drop support for multiple search attribute values for
+non-keywords.
+
+### Minimal proposal
+
+Make the E2E experience consistent, including query construction for listing workflows and search attribute
+registration.
+
+`workflows.ts`
+
+```ts
+import { searchAttributes } from '@temporalio/common';
+
+export const customerAttribute = searchAttributes.defineAttribute(searchAttributes.Text, 'customer');
+export const dateOfPurchaseAttribute = searchAttributes.defineAttribute(Date, 'dateOfPurchase');
```

Comment:
I might drop `define` here. You're not really defining it in the verb sense, especially if it already exists, rather you are referring to some existing definition. Perhaps `attributeDefinition` then?

Reply:
I'd be fine with just `attribute` TBH.

Reply:
I kind of like the verb `define` for consistency with `defineQuery` and `defineSignal`.